pandas/core/internals/blocks.py

```diff
@@ -652,33 +649,11 @@ def astype(self, dtype, copy: bool = False, errors: str = "raise"):
         -------
         Block
         """
-        errors_legal_values = ("raise", "ignore")
-
-        if errors not in errors_legal_values:
-            invalid_arg = (
-                "Expected value of kwarg 'errors' to be one of "
-                f"{list(errors_legal_values)}. Supplied value is '{errors}'"
-            )
-            raise ValueError(invalid_arg)
-
-        if inspect.isclass(dtype) and issubclass(dtype, ExtensionDtype):
-            msg = (
-                f"Expected an instance of {dtype.__name__}, "
-                "but got the class instead. Try instantiating 'dtype'."
-            )
-            raise TypeError(msg)
-
-        dtype = pandas_dtype(dtype)
+        values = self.values
+        if values.dtype.kind in ["m", "M"]:
+            values = self.array_values()
```

Comment:
could move this into astype_array_safe and use ensure_wrapped_if_datetimelike; would make it robust to AM/BM (though i think both AM and BM now  have PRs to make the arrays EAs to begin with)

Reply:
Since ArrayManager already stores it as EAs (after this array), I would prefer to leave it here (then your PR changing to store EAs in BlockManager as well can remove those two lines)